salt/spm/pkgdb/sqlite3.py

```diff
@@ -107,32 +111,42 @@ def list_packages(conn=None):
     '''
     List files for an installed package
     '''
+    close = False
     if conn is None:
+        close = True
         conn = init()
 
     ret = []
     data = conn.execute('SELECT package FROM packages')
     for pkg in data.fetchall():
         ret.append(pkg)
 
+    if close:
+        conn.close()
     return ret
 
 
 def list_files(package, conn=None):
     '''
     List files for an installed package
     '''
+    close = False
     if conn is None:
+        close = True
```

Comment:
You actually can just `None` the `conn` on `.close()` instead of having `close = False | True`:

```python
if conn is not None:
    conn.close()
    conn = None
```

Reply:
Not really, We do not want to close a connection passed in via the `conn` keyword argument.